server/index.js

```diff
@@ -454,17 +413,45 @@ io.on('connection', function(socket) {
             return;
         }
 
-        if(game.players[socket.request.user.username]) {
-            game.players[socket.request.user.username].noReconnect = true;
-        }
-        
-        if(!game.finishedAt) {
-            game.finishedAt = new Date();
-            game.saveGame();
-        }
-
         runAndCatchErrors(game, () => {
-            removePlayerFromGame(game, socket, 'has left the game');
+            if(game.players[socket.request.user.username]) {
+                game.players[socket.request.user.username].noReconnect = true;
```

Comment:
There doesn't seem to be a difference between `Player.left` and `Player.noReconnect` anymore, as they're both always set in this method. Consider removing `Player.noReconnect` and checking `left` instead in the socket connection handler for reconnecting.